src/main/java/com/google/research/bleth/simulator/StrategiesMapper.java

```diff
@@ -0,0 +1,76 @@
+package com.google.research.bleth.simulator;
+
+import org.reflections.Reflections;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A singleton class providing a methods for mapping strings to movement strategy objects and awakeness strategies types.
+ * In addition, the strategies mapper class provides methods for generating a list of strings representing all existing
+ * movement and awakeness strategies, to be used in the new simulation form.
+ */
+public class StrategiesMapper {
+
+    private static StrategiesMapper instance = null;
+    private HashMap<String, Class<? extends IMovementStrategy>> movement = new HashMap<>();
+    private HashMap<String, AwakenessStrategyFactory.Type> awakeness = new HashMap<>();
+
+    /**
+     * A static method to create an instance of the strategies mapper class.
+     * @return the single instance of the mapper class.
+     */
+    public static StrategiesMapper getInstance() {
+        if (instance == null) {
+            instance = new StrategiesMapper();
+        }
+        return instance;
+    }
+
+    /**
+     * Create and return a new movement strategy based on the type provided as a string.
+     * @param typeAsString a string indicating the type of movement strategy to create.
+     * @return a movement strategy object.
+     * @throws NoSuchMethodException if the concrete movement strategy does not have a declared constructor.
+     * @throws IllegalAccessException if the concrete movement strategy's constructor is not accessible.
+     * @throws InstantiationException if the instantiation fails.
+     */
+    public IMovementStrategy getMovementStrategy(String typeAsString)
+            throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        return movement.get(typeAsString).getDeclaredConstructor().newInstance();
```

Comment:
Works for now, won't work when we add the pre-defined strategy.

Reply:
I agree. Will be refactored as soon as we will have such a strategy.